src/wallet/components/ConnectWallet.tsx

```diff
@@ -80,7 +96,16 @@ export function ConnectWallet({
         <ConnectButton
           className={className}
           connectWalletText={connectWalletText}
-          onClick={() => connect({ connector })}
+          onClick={() => {
+            connect(
+              { connector },
+              {
+                onSuccess: () => {
+                  onInitialConnect?.();
+                },
```

Comment:
Safer, more reliable way of triggering the callback. We have to use the useEffect and state for RainbowKit support